LSP not working on Linux

### Environment

* **OS**:  Arch linux
* **Version**: 0.3.1

### Problem
LSP not working in linux  because the wrong path i think look at the log:

```
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: node:internal/modules/cjs/loader:1424
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: throw err;
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: ^
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: Error: Cannot find module '/usr/lib/Athas/bundled/typescript/lsp/node_modules/typescript-language-server/lib/cli.mjs'
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at Module._resolveFilename (node:internal/modules/cjs/loader:1421:15)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at defaultResolveImpl (node:internal/modules/cjs/loader:1059:19)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1064:22)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at Module._load (node:internal/modules/cjs/loader:1227:37)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at TracingChannel.traceSync (node:diagnostics_channel:328:14)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at wrapModuleLoad (node:internal/modules/cjs/loader:245:24)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at Module.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:154:5)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at node:internal/main/run_main_module:33:47 {
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: code: 'MODULE_NOT_FOUND',
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: requireStack: []
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: }
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: Node.js v24.11.1
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: node:internal/modules/cjs/loader:1424
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: throw err;
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: ^
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: Error: Cannot find module '/usr/lib/Athas/bundled/typescript/lsp/node_modules/typescript-language-server/lib/cli.mjs'
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at Module._resolveFilename (node:internal/modules/cjs/loader:1421:15)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at defaultResolveImpl (node:internal/modules/cjs/loader:1059:19)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1064:22)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at Module._load (node:internal/modules/cjs/loader:1227:37)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at TracingChannel.traceSync (node:diagnostics_channel:328:14)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at wrapModuleLoad (node:internal/modules/cjs/loader:245:24)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at Module.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:154:5)
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: at node:internal/main/run_main_module:33:47 {
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: code: 'MODULE_NOT_FOUND',
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: requireStack: []
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: }
[03:10:44AM] [ERROR] [athas_code::lsp::client]: LSP stderr: Node.js v24.11.1
[03:10:44AM] [ERROR] [athas_code::commands::development::lsp]: Failed to get completions: No LSP client for this file

```

the current path to `bundled` directory is `/usr/lib/Athas/_up_/src/extensions/bundled/`

---

## Checklist

* [x] I have searched the existing issues
* [x] I have reproduced this issue with the latest version

--- a/src-tauri/Cargo.toml
+++ b/src-tauri/Cargo.toml
@@ -42,7 +42,7 @@ serde = { version = "1.0", features = ["derive"] }
 sha2 = "0.10"
 serde_json = "1.0"
 ssh2 = { version = "0.9", features = ["vendored-openssl"] }
-tauri = { version = "2", features = ["protocol-asset", "macos-private-api", "unstable"] }
+tauri = { version = "2", features = ["protocol-asset", "macos-private-api", "unstable", "test"] }
 tauri-plugin-dialog = "2"
 tauri-plugin-fs = "2"
 tauri-plugin-http = "2"

--- a/src-tauri/src/commands/extensions.rs
+++ b/src-tauri/src/commands/extensions.rs
@@ -6,7 +6,7 @@ use std::{
    io::Write,
    path::{Path, PathBuf},
 };
-use tauri::{AppHandle, command};
+use tauri::{AppHandle, Runtime, command};
 
 #[command]
 pub async fn download_extension(
@@ -162,7 +162,7 @@ fn get_extensions_dir() -> Result<PathBuf, String> {
 }
 
 #[command]
-pub fn get_bundled_extensions_path(app_handle: AppHandle) -> Result<String, String> {
+pub fn get_bundled_extensions_path<R: Runtime>(app_handle: AppHandle<R>) -> Result<String, String> {
    // In production, use Tauri's resource directory API
    // In development, fall back to the source path
    let extensions_path = if cfg!(debug_assertions) {
@@ -262,4 +262,81 @@ pub fn get_extension_path(app_handle: AppHandle, extension_id: String) -> Result
       .to_str()
       .ok_or("Failed to convert path to string")?
       .to_string())
+}
+
+#[cfg(test)]
+mod tests {
+   use super::*;
+   use std::path::Path;
+
+   #[test]
+   fn test_get_bundled_extensions_path_ends_with_bundled() {
+      // Create a mock Tauri app for testing
+      let app = tauri::test::mock_app();
+      let app_handle = app.handle().clone();
+
+      // Call the function
+      let result = get_bundled_extensions_path(app_handle);
+
+      // Verify it succeeds and the path ends with "bundled"
+      assert!(result.is_ok(), "get_bundled_extensions_path should succeed");
+      let path = result.unwrap();
+      let path = Path::new(&path);
+
+      // The path must end with "bundled", not "_up_/src/extensions/bundled"
+      // This verifies the fix for issue #475 where Linux builds had wrong paths
+      assert!(
+         path.ends_with("bundled"),
+         "Path should end with 'bundled', got: {:?}",
+         path
+      );
+
+      // Verify the path doesn't contain "_up_" which indicates incorrect Tauri resource bundling
+      assert!(
+         !path.to_string_lossy().contains("_up_"),
+         "Path should not contain '_up_' (incorrect bundling), got: {:?}",
+         path
+      );
+   }
+
+   #[test]
+   fn test_get_bundled_extensions_path_is_absolute_in_debug() {
+      let app = tauri::test::mock_app();
+      let app_handle = app.handle().clone();
+
+      let result = get_bundled_extensions_path(app_handle);
+      assert!(result.is_ok());
+
+      let path_str = result.unwrap();
+      let path = Path::new(&path_str);
+
+      // In debug mode, the path should be constructed from current_dir
+      // and should be an absolute path
+      assert!(
+         path.is_absolute(),
+         "Path should be absolute in debug mode, got: {:?}",
+         path
+      );
+   }
+
+   #[test]
+   fn test_get_bundled_extensions_path_contains_expected_structure() {
+      let app = tauri::test::mock_app();
+      let app_handle = app.handle().clone();
+
+      let result = get_bundled_extensions_path(app_handle);
+      assert!(result.is_ok());
+
+      let path_str = result.unwrap();
+
+      // In debug mode, path should contain src/extensions/bundled
+      // This is the development path structure
+      assert!(
+         path_str.contains("src")
+            && path_str.contains("extensions")
+            && path_str.ends_with("bundled"),
+         "Debug path should have structure .../src/extensions/bundled, got: {}",
+         path_str
+      );
+   }
 }

--- a/src-tauri/tauri.conf.json
+++ b/src-tauri/tauri.conf.json
@@ -57,9 +57,9 @@
       "icons/icon.icns",
       "icons/icon.ico"
     ],
-    "resources": [
-      "../src/extensions/bundled"
-    ],
+    "resources": {
+      "../src/extensions/bundled": "bundled"
+    },
     "macOS": {
       "frameworks": [],
       "minimumSystemVersion": "10.15",

